src/groundlight/client.py

```diff
@@ -347,11 +359,134 @@ def submit_image_query(  # noqa: PLR0913 # pylint: disable=too-many-arguments
             iq_id = self.api_client.submit_image_query_with_inspection(**params)
             image_query = self.get_image_query(iq_id)
 
-        if wait:
-            threshold = self.get_detector(detector).confidence_threshold
+        if wait > 0:
+            if confidence_threshold is None:
+                threshold = self.get_detector(detector).confidence_threshold
+            else:
+                threshold = confidence_threshold
```

Comment:
The next line is in the if statement too. If wait is set to 0 (or less) then we're not going to make a call to `wait_for_confident_result` and since we don't make that call we don't need the threshold value.